### PR TITLE
Add API for parsed JSON

### PR DESCRIPTION
**Pull request by klauspost**
2 tests fail:

* Numbers mismatch as discussed.
* TestDumpRawDemoJson - could be something on Windows.

Not sure if `Tape` and `Strings` should be exported in ParsedJson.

Also seems like gofmt/goimports disagrees on some parts . Gofmt your stuff ;)

Marshal Speed (output MB/s):
```
BenchmarkIter_MarshalJSONBuffer/apache_builds-12         	    4614	    252773 ns/op	 374.46 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/canada-12                	      85	  14580301 ns/op	 142.99 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/citm_catalog-12          	     820	   1475940 ns/op	 338.97 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/github_events-12         	    9228	    132606 ns/op	 402.16 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/gsoc-2018-12             	     186	   6479951 ns/op	 473.78 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/instruments-12           	    3332	    355792 ns/op	 304.43 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/numbers-12               	    1062	   1153368 ns/op	 136.97 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/marine_ik-12             	      74	  15922568 ns/op	 140.61 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/mesh-12                  	     274	   4336062 ns/op	 154.81 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/mesh.pretty-12           	     271	   4430731 ns/op	 153.67 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/twitterescaped-12        	     982	   1218988 ns/op	 383.04 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/twitter-12               	     991	   1207141 ns/op	 386.80 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/random-12                	     838	   1415048 ns/op	 326.11 MB/s	     192 B/op	       2 allocs/op
BenchmarkIter_MarshalJSONBuffer/update-center-12         	     832	   1445577 ns/op	 368.68 MB/s	     192 B/op	       2 allocs/op
```

Using stdlib json.Marshal:
```
BenchmarkGoMarshalJSON/apache_builds-12         	     997	   1166832 ns/op	  81.24 MB/s	  564542 B/op	   10593 allocs/op
BenchmarkGoMarshalJSON/canada-12                	      45	  25983624 ns/op	  80.44 MB/s	 2098820 B/op	      39 allocs/op
BenchmarkGoMarshalJSON/citm_catalog-12          	      99	  12760356 ns/op	  39.21 MB/s	 5464813 B/op	  117349 allocs/op
BenchmarkGoMarshalJSON/github_events-12         	    2352	    520100 ns/op	 102.65 MB/s	  209204 B/op	    3337 allocs/op
BenchmarkGoMarshalJSON/gsoc-2018-12             	     100	  10802505 ns/op	 284.40 MB/s	 5759292 B/op	   60679 allocs/op
BenchmarkGoMarshalJSON/instruments-12           	     440	   2727119 ns/op	  39.72 MB/s	  974510 B/op	   18837 allocs/op
BenchmarkGoMarshalJSON/numbers-12               	     681	   1768384 ns/op	  84.89 MB/s	  155650 B/op	       1 allocs/op
BenchmarkGoMarshalJSON/marine_ik-12             	      24	  49436196 ns/op	  37.12 MB/s	 7677456 B/op	  134427 allocs/op
BenchmarkGoMarshalJSON/mesh-12                  	     100	  11072398 ns/op	  58.05 MB/s	  648816 B/op	      37 allocs/op
BenchmarkGoMarshalJSON/mesh.pretty-12           	     100	  11063652 ns/op	  58.09 MB/s	  648816 B/op	      37 allocs/op
BenchmarkGoMarshalJSON/twitterescaped-12        	     189	   6191865 ns/op	  76.06 MB/s	 2075920 B/op	   33583 allocs/op
BenchmarkGoMarshalJSON/twitter-12               	     194	   6169387 ns/op	  76.34 MB/s	 2075920 B/op	   33583 allocs/op
BenchmarkGoMarshalJSON/random-12                	     139	   8618876 ns/op	  53.54 MB/s	 3363574 B/op	   64015 allocs/op
BenchmarkGoMarshalJSON/update-center-12         	     187	   6402509 ns/op	  83.96 MB/s	 2412632 B/op	   40627 allocs/op
```

### CODE REVIEW

**Comment by klauspost** [comment]
Number precision seems to be discussed here: https://github.com/lemire/simdjson/issues/242

**Comment by klauspost** [comment]
And PR here: https://github.com/lemire/simdjson/pull/303 👍

File: parsed_json.go [hunk 1]
```diff
@@ -2,167 +2,1044 @@ package simdjson
...
+		i.off++
+		return float64(v), nil
+	default:
+		return 0, fmt.Errorf("unable to convert type %v to float", i.t)
+	}
+}
 
-		case 't': // we have a true
-			fmt.Printf("true")
+// Float returns the float value of the next element.
```
Comment (fwessels) [inline parsed_json.go:335]: Comment is incorrect

File: parsed_json.go [hunk 2]
```diff
@@ -2,167 +2,1044 @@ package simdjson
...
+		}
+		return int64(v), nil
+	default:
+		return 0, fmt.Errorf("unable to convert type %v to float", i.t)
+	}
+}
 
-		case 'f': // we have a false
-			fmt.Printf("false")
+// Float returns the float value of the next element.
```
Comment (fwessels) [inline parsed_json.go:374]: Same

File: parsed_json.go [hunk 3]
```diff
@@ -2,167 +2,1044 @@ package simdjson
...
+		return "", errors.New("value is not string")
+	}
+	return i.tape.StringAt(i.cur)
+}
 
-		case '[': // we start an array
-			fmt.Printf("[")
-			inobject = append(inobject, false)
-			inobjectidx = append(inobjectidx, 0)
+// String() returns a string value.
```
Comment (fwessels) [inline parsed_json.go:422]: Idem

File: parsed_json.go [hunk 4]
```diff
@@ -2,167 +2,1044 @@ package simdjson
...
-			return false
+			return nil, fmt.Errorf("unable to convert type %v to float", tag)
 		}
+		a.off++
 	}
+	return dst, nil
+}
 
-	return true
+// AsFloat returns the array values as float.
```
Comment (fwessels) [inline parsed_json.go:843]: AsInteger

File: stage1_find_marks_test.go [hunk 5]
```diff
@@ -39,10 +39,10 @@ func TestStage1FindMarks(t *testing.T) {
 	}{
 		{
 			// {"Image":{"Width":800,"Height":600,"Title":"View from 15th Floor
-			  "0111111000111111000000111111100000011111100111111111111111111111", // quoted
-			  "1000000011000000010001000000001000100000001000000000000000000000", // structurals
-			  "0000000000000000000000000000000000000000000000001000010000100000", // whitespace
-			  "1100000011100000011001100000001100110000001100000000000000000000", // structurals_finalized
+			"0111111000111111000000111111100000011111100111111111111111111111", // quoted
```
Comment (fwessels) [inline stage1_find_marks_test.go:42]: These are intentionally aligned with their corresponding character above...

Comment (klauspost) [reply stage1_find_marks_test.go:42]: yeah, but gofmt disagrees 😢

**Pull request merged**